src/radical/pilot/session.py

```diff
@@ -377,32 +377,23 @@ def _init_cfg_from_scratch(self):
                                                        cfg=self._cfg)
 
         rcfgs = ru.Config('radical.pilot.resource', name='*', expand=False)
-        rcfgs_ext = {}
+        self._rcfgs = ru.Config()
 
         for site in rcfgs:
-            rcfgs_ext[site] = {}
+            self._rcfgs[site] = ru.Config()
             for res, rcfg in rcfgs[site].items():
-                rcfgs_ext[site][res] = {
-                    'default_schema': rcfg['default_schema'],
-                    'schemas'       : rcfg.get('schemas', {})
-                }
-                for schema in rcfg.get('schemas', {}):
-                    while isinstance(rcfg['schemas'][schema], str):
-                        tgt = rcfg['schemas'][schema]
-                        rcfg['schemas'][schema] = rcfg['schemas'][tgt]
-                for schema in rcfg.get('schemas', {}):
-                    rcfgs_ext[site][res][schema] = rcfgs[site][res].as_dict()
-                    ru.dict_merge(rcfgs_ext[site][res][schema],
-                                  rcfgs[site][res]['schemas'][schema])
-                    del rcfgs_ext[site][res][schema]['default_schema']
-
-        for site in rcfgs_ext:
-            for res, rcfg in rcfgs_ext[site].items():
-                for schema in rcfg.get('schemas', {}):
-                    rd = ResourceDescription(from_dict=rcfg[schema])
-                    rd.verify()
-
-        self._rcfgs = ru.Config(from_dict=rcfgs_ext)
+                self._rcfgs[site][res] = ResourceConfig(rcfg)
+
+                # resolve schema aliases
+                for schema, tgt in rcfg.get('schemas', {}).items():
+                    alias = False
+                    while isinstance(tgt, str):
+                        alias = True
+                        tgt = rcfg['schemas'].get(tgt)
+                    if alias:
+                        rcfg['schemas'][schema] = tgt
```

Comment:
we can keep an original one
```python
                for schema in rcfg.get('schemas', {}):
                    while isinstance(rcfg['schemas'][schema], str):
                        tgt = rcfg['schemas'][schema]
                        rcfg['schemas'][schema] = rcfg['schemas'][tgt]
```
thus we'll get KeyError if `alias` refers to non-existing schema

Reply:
After discussion: remove alias support

Reply:
Done